src/Security/SessionDownloadStrategy.php

```diff
@@ -58,24 +59,43 @@ class SessionDownloadStrategy implements DownloadStrategyInterface
      * @param ContainerInterface|SessionInterface $session
      * @param int                                 $times
      */
-    public function __construct(TranslatorInterface $translator, $session, $times)
+    public function __construct(object $translator, $session, $times)
     {
+        if ($translator instanceof LegacyTranslatorInterface) {
+            @trigger_error(sprintf(
+                'Passing other type than "%s" as argument 1 to "%s()" is deprecated since sonata-project/media-bundle 3.x'
+                .' and will throw a "\TypeError" error in 4.0.',
+                TranslatorInterface::class,
+                __METHOD__
+            ), \E_USER_DEPRECATED);
+        } elseif (!$translator instanceof TranslatorInterface) {
+            throw new \TypeError(sprintf(
+                'Argument 1 passed to "%s()" MUST be an instance of "%s" or "%s", "%s" given.',
+                __METHOD__,
+                LegacyTranslatorInterface::class,
+                TranslatorInterface::class,
+                \get_class($translator)
+            ));
+        }
+
         // NEXT_MAJOR : remove this block and set session from parameter.
         if ($session instanceof ContainerInterface) {
-            @trigger_error(
-                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since
+            @trigger_error(sprintf(
+                'Using an instance of "%s" is deprecated since
                 version 3.1 and will be removed in 4.0.
-                Use Symfony\Component\HttpFoundation\Session\SessionInterface instead.',
-                \E_USER_DEPRECATED
-            );
+                Use "%s" instead.',
+                ContainerInterface::class,
+                SessionInterface::class
+            ), \E_USER_DEPRECATED);
 
             $this->session = $session->get('session');
         } elseif ($session instanceof SessionInterface) {
             $this->session = $session;
         } else {
-            throw new \InvalidArgumentException(
-                '$session should be an instance of Symfony\Component\HttpFoundation\Session\SessionInterface'
-            );
+            throw new \InvalidArgumentException(sprintf(
+                '$session should be an instance of "%s"',
+                SessionInterface::class
```

Comment:
```suggestion
            throw new \TypeError(sprintf(
                'Argument 2 passed to "%s()" MUST be an instance of "%s" or "%s", "%s" given.',
                __METHOD__,
                SessionInterface::class,
                ContainerInterface::class,
                \get_class($session)
```

Reply:
We MUST keep `\InvalidArgumentException`. Message are updated.